.github/workflows/test.yml

```diff
@@ -22,32 +26,47 @@ jobs:
         python-version: ${{ matrix.python-version }}
 
     - name: Install Poetry
-      run: pip install poetry==1.1.15
-
-    - name: Determine poetry version
-      run: echo "::set-output name=VERSION::$(poetry --version)"
-      id: poetry_version
-
-    - name: Cache poetry.lock
-      uses: actions/cache@v3
+      uses: snok/install-poetry@v1
       with:
-        path: poetry.lock
-        key: ${{ matrix.os }}-${{ matrix.python-version }}-poetry-${{ steps.poetry_version.outputs.VERSION }}-${{ hashFiles('pyproject.toml') }}
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+  
+    - name: Load cached venv
+      id: cached-pip-wheels
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache
+        key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
 
     - name: Install project dependencies
       run: poetry install -vvv --no-root
 
     - name: Run mypy on 'tests' (using the local stubs) and on the local stubs
-      run: poetry run poe mypy
+      run: |
+        source $VENV
+        poetry run poe mypy
 
     - name: Run pyright on 'tests' (using the local stubs) and on the local stubs
-      run: poetry run poe pyright
+      run: |
+        source $VENV
+        poetry run poe pyright
 
     - name: Run pytest
-      run: poetry run poe pytest
+      run: |
+        source $VENV
+        poetry run poe pytest
 
     - if: matrix.python-version == '3.8' && matrix.os == 'ubuntu-latest'
       uses: pre-commit/action@v3.0.0
 
     - name: Install pandas-stubs and run tests on the installed stubs
-      run: poetry run poe test_dist
+      run: |
+        source $VENV
```

Comment:
In the above steps we can use `poe ...`, why can't we use it here?

Reply:
As mentioned above, if I use the run script, it installs the wheel into the "system" environment, not poetry's virtual environment.